### ISSUE TITLE
[POLL] How much JMMServer use in IDLE mode?

### ISSUE THREAD

**Issue by misakitchi** [japanesemediamanager/ShokoServer#155]
Normally JMMServer must use 0% CPU in IDLE mode

IDLE mode: NO queue (Hasher=0 General=0 Images=0) and JMMDesktop closed

But me i have a very annoying bug!
JMMServer use all of my CPU! :(

So I create a Straw Poll to know how many people have this bug too:

http://strawpoll.me/4955543/

You can use "Process Explorer" and watch in tab "Performance Graph" to know how many CPU is used.

Thx for help!

PS: Process Explorer link:
https://technet.microsoft.com/en-us/sysinternals/bb842062

**Comment by AreFinstad** [comment]
always above 10% for me, at times it stays at 15% way too much considering the CPU i have here.

**Comment by gibwar** [comment]
0% here and 91,584KB used. Uptime of 5 days. Previous uptime of 45 days had an average of 1% CPU. Machine has 4 cores given to it, i7-3615QM (2.3 GHz), Windows 8.1 VM running on a Mac Mini. Monitors an iSCSI target for stuff to import and moves it to a normal SMB share (no direct monitoring on SMB share).

**Comment by AreFinstad** [comment]
@gibwar considering i got a i7 3960x as a hexa core to use that much 24/7 is too much IMO. ram usage idle around 135mb aswell.

**Comment by Animeimp** [comment]
0% CPU, 7808kb memory used.  It has been up and running for 8 days.  Main database is Microsoft SQL 2012 on the same machine.  Anime is stored on a Synology NAS device, with the folders mapped to drive letters.  Currently in the database is 20,706 videos, taking up 6319.37 GB.

TrakTV, MovieDB and The TvDB is turned off, only Anidb being used for data pulls.

**Comment by jmediamanager** [comment]
Yes it would be good if everyone can give more details about their system setup, this might help narrow it down.
For those have already posted, maybe you could go back and edit your post

**CPU** -  0%
**Memory** -  168MB
**Database Type** - SQL Server 2008
**Collection Size** - 1300 Series
**Trakt/TvDB/MovieDB** -  On / On / On
**Auto Sync Mylist** - Off
**Auto Sync Stats** - Off
**Windows Regional Language** - English

**Comment by AreFinstad** [comment]
updated comment for specs.... yay

**Comment by jmediamanager** [comment]
Yes the region could possibly matter. It can cause all sorts of weird issues, if I haven't written the code correctly.

Also, if anyone is seeing JMM Server using more than 0% CPU, I would recommend changing the following in the config file.

`<logger name="*" minlevel="Info" writeTo="file,console"/>`
to
`<logger name="*" minlevel="Trace" writeTo="file,console"/>`

Then check to logs to see if JMM Server is actually logging anything, when you think it should be doing nothing

**Comment by maxpiva** [comment]
JMM Server IDLE 0%, Memory: 173MB
SQL Server 2014 IDLE 0%, Memory: 770MB
Collection size: 3658 Series
*Trakt/TvDB/MovieDB* - On  / On / On
*Auto Sync Mylist* - On
*Auto Sync Stats* - On
*Windows Regional Language* - English

**Comment by misakitchi** [comment]
OK i put my settings too:

System:
OS: Win 7
CPU: Intel Pentium E5300@3.46GHz
RAM: 6Go DDR2
Database Type: MS SQL 2014
Windows Region: French
Display Language: French
JMM Server IDLE: 40-50% CPU, Memory: 600Mo-1.2Go

**Comment by ElementalCrisis** [comment]
CPU - 0%
Memory - 111.7MB
Database Type - MySQL Community Server 5.6.25
Collection Size - 674 Series
Trakt/TvDB/MovieDB - Off / On / On
Auto Sync Mylist - On
Auto Sync Stats - On
Windows Regional Language - English

**Comment by bond4u** [comment]
There seems to be some magic number X of unrecognized files:
bellow X - 0% cpu and ~200mb memory usage
above X - 25% cpu and 1g memory usage
And another magic number Y:
above Y - 40% cpu and 3g memory usage.
win7 and mysql
jmmserver 3.4.3.2 & 3.4.4.3

**Comment by AreFinstad** [comment]
well the program isnt that smooth for everyone, not for me atleast.
i still see around 15-20% CPU usage even with empty queues and it stays like that till it i shut down the server, meaning it can stay like that for more than a day sometimes.

will have to get some logs about this with it running for a few hours....

**Comment by AreFinstad** [comment]
log time!

here: https://drive.google.com/file/d/0B-hww4iUE6HsblAwMEdBT0NaS0E/view?usp=sharing

i have let the server run for some time here and usage is still really high, never bellow 10% and goes up and down all the time (mostly up) 
hope this helps somehow.
the queues where empty when i started aswell as now when i upload, tho it does look like some things where updated this night.

**Comment by misakitchi** [comment]
AreFin did you do the command: "delete from LogMessage" in SQL Server Management Studio?
I do this command each month and now JMM use 0% when idle :)

**Comment by AreFinstad** [comment]
ehhhh nope, havent touched anything on the MySQL side at all since installing it for JMM

**Comment by misakitchi** [comment]
I must do that every month... but now JMM use 0% in idle :)
But now i search a way to automatise this command, have not found a solution yet... :(

**Comment by ElementalCrisis** [comment]
Create a new thread with that feature request, best way for @jmediamanager to keep track of ideas.

**Comment by bigretromike** [comment]
0% CPU and 200MB using MySQL

**Comment by ElementalCrisis** [comment]
Closing this issue as recent work has reduced the memory usage of Shoko Server.

**Comment by bigretromike** [comment]
reduce? This issue was before we started Caching database. So it only increased.

**Comment by ElementalCrisis** [comment]
We're never going to get down to pre-cache levels. Recent changes have made it better, currently sitting at 15MB Idling and 200-300MB when hashing which is a lot better than my previous results.

**Comment by da3dsoul** [comment]
This isn't exactly accurate to begin with, as people aren't counting SQL ram usage. I already got Desktop down. Server uses ram, it always has, and always will. We moved the RAM from SQL to server, so now it looks like it's using more. My collection uses 700mb of RAM. About 10% is wasted according to dotMemory. That's still 630mb, and I don't really care. I have 32GB, and I'll gladly give Shoko 1 or 2 of it
